openstack_controller/datadog_checks/openstack_controller/config.py

```diff
@@ -0,0 +1,128 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+
+from openstack.config import loader
+
+from datadog_checks.base import ConfigurationError
+from datadog_checks.base.utils.models.types import copy_raw
+from datadog_checks.openstack_controller.api.type import ApiType
+from datadog_checks.openstack_controller.defaults import DEFAULT_DOMAIN_ID
+
+
+# Discovery class requires 'include' to be a dict, so this function is needed to normalize the config
+def normalize_discover_config_include(config, item_keys):
+    normalized_config = {}
+    include_list = config.get('include') if isinstance(config, dict) else copy_raw(config.include) if config else []
+    if include_list:
+        if not isinstance(include_list, list):
+            raise TypeError('Setting `include` must be an array')
+        for entry in include_list:
+            if isinstance(entry, str):
+                normalized_config[entry] = None
+            elif isinstance(entry, dict):
+                dict_key = None
+                for key in item_keys:
+                    if key in entry.keys():
+                        normalized_config[entry[key]] = entry
+                        break
+                if dict_key:
+                    normalized_config[dict_key] = entry
+            else:
+                raise TypeError('`include` entries must be a map or a string')
+    return normalized_config
+
+
+class OpenstackConfig:
+    def __init__(self, logger, instance):
+        self.log = logger
+        self.instance = instance
+        self.openstack_config_file_path = instance.get("openstack_config_file_path")
+        self.openstack_cloud_name = instance.get("openstack_cloud_name")
+        self.keystone_server_url = instance.get("keystone_server_url")
+        self.username = instance.get("username")
+        self.password = instance.get("password")
+        self.domain_id = instance.get("domain_id", DEFAULT_DOMAIN_ID)
+        self.user = instance.get("user")
+        self.nova_microversion = instance.get('nova_microversion')
+        self.ironic_microversion = instance.get('ironic_microversion')
```

Comment:
I think there's a default version declared in the config file https://github.com/DataDog/integrations-core/pull/15918/files#diff-d687126414da4b2b93d1f38c6d0c7cd1b6f564463dbbfcebd237ec2108ce4c06R64